sui_programmability/adapter/src/adapter.rs

```diff
@@ -527,8 +529,12 @@ pub fn generate_package_id(
         let old_module_id = module.self_id();
         let old_address = *old_module_id.address();
         if old_address != AccountAddress::ZERO {
+            let handle = module.module_handle_at(module.self_module_handle_idx);
+            let name = module.identifier_at(handle.name);
             return Err(SuiError::ModulePublishFailure {
-                error: "Publishing modules with non-zero address is not allowed".to_string(),
+                error: format!(
+                    "Publishing modules with non-zero address is not allowed (module {name})"
```

Comment:
```suggestion
                    "Publishing module {name} with non-zero address is not allowed."
```

Reply:
I wanted to avoid modifying tests, but it was silly... Fixed!